base_layer/core/src/transactions/transaction_protocol/transaction_initializer.rs

```diff
@@ -214,6 +214,13 @@ impl SenderTransactionInitializer {
     ) -> Result<&mut Self, BuildError> {
         let commitment_factory = PedersenCommitmentFactory::default();
         let commitment = commitment_factory.commit(&output.spending_key, &PrivateKey::from(output.value));
+        let recovery_byte = OutputFeatures::create_unique_recovery_byte(&commitment, self.rewind_data.as_ref());
```

Comment:
Feels like this work has already been done. Why do we not add the commitment into the unblindedoutput and skip recreating it here?

Reply:
I think it has to do with the commitment factory that is needed to calculate the commitment, which will make creating an unblinded output a heavier operation.

In this example `commitment ` is needed for the metadata signature further on; it is just re-used here for the recovery byte calculation.